spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/integration/IntegrationProperties.java

```diff
@@ -295,4 +302,86 @@ public void setMessageMappingEnabled(boolean messageMappingEnabled) {
 
 	}
 
+	public static class Poller {
+
+		/**
+		 * Maximum of messages to poll per polling cycle.
+		 */
+		private Long maxMessagesPerPoll;
+
+		/**
+		 * How long to wait for messages on poll.
+		 */
+		private Duration receiveTimeout;
+
+		/**
+		 * Polling delay period.
+		 */
+		private Duration fixedDelay;
+
+		/**
+		 * Polling rate period.
+		 */
+		private Duration fixedRate;
+
+		/**
+		 * Polling initial delay.
+		 */
+		private Duration initialDelay;
+
+		/**
+		 * Cron expression for polling.
```

Comment:
It would be good to document that `cron`, `fixedRate`, and `fixedDelay` are mutually exclusive.

Reply:
This is true. 
I Stream Applications projects we use `@Validated` feature to prevent ambiguity downstream and to avoid precedence as you pointed in the next comment: https://github.com/spring-projects/spring-boot/pull/27992#discussion_r709106055.
The approach is like this: https://github.com/spring-cloud/stream-applications/blob/main/functions/consumer/log-consumer/src/main/java/org/springframework/cloud/fn/consumer/log/LogConsumerProperties.java#L57.
Since I don't see anything like that in Spring Boot by itself, I forgot to ask such a question when opened PR.

Thank you for reminder!
So, now let's discuss what is the best way to go.
I simply can validate exclusiveness in target code where I apply those properties, although it is still feels more nature to validate them on the configuration properties phase.
If you point me to something similar, I'll fix it ASAP

Reply:
I've labeled this for discussion at a team meeting so that we can consider our options.

Reply:
We're going to follow the precedent set by Framework's handling of `@Scheduled` and fail if more than one of `cron`, `fixed-rate`, and `fixed-delay` is set.

Reply:
Doing it in the `PollerMetadata` `@Bean` method sounds good to me. Generally, we try to keep the properties classes themselves as free of logic as possible.

>Although it is not clear why you are talking about precedence

I was talking about the precedent set by Framework in how it handles similar properties on `@Scheduled`. Nothing to do with which property takes precedence over another. [That was an option](https://github.com/spring-projects/spring-boot/pull/27992#discussion_r709106055), but we've decided to fail now.